data/scripts.js

```diff
@@ -636,14 +636,7 @@ exports.BattleScripts = {
 		return isValid;
 	},
 	sampleNoReplace: function (list) {
-		let length = list.length;
-		let index = this.random(length);
-		let element = list[index];
-		for (let nextIndex = index + 1; nextIndex < length; index += 1, nextIndex += 1) {
```

Comment:
Next thing you'll be telling me that `index += 1` is faster than `++index`... wait, yes, it's 3.5% faster; now I've seen it all...